app/frontend/src/features/profile/edit/EditHostingPreference.tsx

```diff
@@ -250,31 +252,25 @@ export default function HostingPreferenceForm() {
             defaultValue={user.smokingAllowed}
             name="smokingAllowed"
             render={({ onChange }) => (
-              <Autocomplete
-                disableClearable={false}
-                defaultValue={user.smokingAllowed}
-                forcePopupIcon
-                freeSolo={false}
-                getOptionLabel={(option) => smokingLocationLabels[option]}
-                multiple={false}
-                options={[
+              <Select
+                defaultValue={user.smokingAllowed || SmokingLocation.SMOKING_LOCATION_UNKNOWN}
```

Comment:
Seems like code smell to have `defaultValue` on both the controller and the rendered component. Maybe the most correct way would be to set `value` on the component from `render={({ onChange, value }) => (` ? Not totally sure

Reply:
yes :( seems it didn't work indeed. I've added tests now to check if the correct value from the database is set in the form.